### ISSUE TITLE
Warnings vs ValueErrors with user provided params

### ISSUE THREAD

**Issue by VeckoTheGecko** [OceanPARCELS/parcels#1691]
We have some warnings along the lines of 

https://github.com/OceanParcels/Parcels/blob/5fff42c2ea4cc12d03df8c936dd391e251658c8f/parcels/field.py#L232-L238

Thoughts on just raising a ValueError instead? Its incorrect usage of the API, and I think explicitly failing would be nicer than implicitly working under different settings

**Comment by VeckoTheGecko** [comment]
I don't think its a major thing either way. Perhaps worth it only for v4

**Comment by erikvansebille** [comment]
Yes, good point that a `ValueError` would be more appropriate and cleaner here; the code then also don't need to implicitly decide that `time_periodic` is a more important setting than `allow_time_interpolation`
